chemprop/uncertainty/evaluator.py

```diff
@@ -41,14 +44,68 @@ def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -
 
 @UncertaintyEvaluatorRegistry.register("miscalibration_area")
 class CalibrationAreaEvaluator(RegressionEvaluator):
+    """
+    A class for evaluating regression uncertainty values based on how they deviate from perfect
+    calibration on an observed-probability versus expected-probability plot.
+    """
+
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        return
+        bin_scaling = torch.tensor([erfinv(i / 100) * np.sqrt(2) for i in range(1, 100)])
+        bin_scaling = bin_scaling.view(-1, 1, 1)
+        errors = torch.abs(preds - targets)
+        uncs = torch.sqrt(uncs).unsqueeze(0)
+        bin_unc = uncs * bin_scaling
+        bin_count = bin_unc >= errors.unsqueeze(0)
+        mask = mask.unsqueeze(0)
+        bin_fraction = (bin_count & mask).sum(1) / mask.sum(1)
+        fractions = torch.zeros((uncs.shape[-1], 101))
+        fractions[:, 100] = 1.0
+        fractions[:, 1:100] = bin_fraction.permute(1, 0)
+        miscal_area = torch.sum(0.01 * torch.abs(fractions - torch.arange(101) / 100), dim=1)
+        return miscal_area
 
 
 @UncertaintyEvaluatorRegistry.register("ence")
 class ExpectedNormalizedErrorEvaluator(RegressionEvaluator):
+    r"""
+    A class that evaluates uncertainty performance by binning together clusters of predictions
+    and comparing the average predicted variance of the clusters against the RMSE of the cluster. [1]_
+
+    .. math::
+        ENCE = \frac{1}{K} \sum_{i=1}^{K} \frac{|\text{mVAR}(i) - RMSE(i)|}{\text{mVAR}(i)}
```

Comment:
Yes, mVAR and RMSE are two lists. I am following Equation 11 from the paper. Maybe we can change it to:
```suggestion
        \text{ENCE} = \frac{1}{K} \sum_{i=1}^{K} \frac{|\mathrm{mVAR}_i - \mathrm{RMSE}_i|}{\mathrm{mVAR}_i}
```

Reply:
can we just use $\sigma_i$ and $e$ for mVAR and RMSE?

Reply:
Actually, the original paper of this method presents the root of the mean variance and root mean square error for each bin as as function, where $|B_i|$ is the size of bin $i$. How about I add these two equations above the ENCE definition and we keep the current style?
$RMV(j) = \sqrt{\frac{1}{|B_j|} \sum_{t \in B_j} \sigma_t^2}$
$RMSE(j) = \sqrt{\frac{1}{|B_j|} \sum_{t \in B_j} \left( y_t - \hat{y}_t \right)^2}$

Reply:
IMO it should still be index notation: $\text{RMV}_i$ and $\text{RMSE}_i$

Reply:
I agree with you. I would prefer to use $\text{RMV}_i$ and $\text{RMSE}_i$ instead of $\sigma_i$ and $e_i$ as it would be easier to match the source of the paper.